core/controllers/admin_test.py

```diff
@@ -128,12 +128,12 @@ def test_change_rights(self):
         self.signup(
             BOTH_MODERATOR_AND_ADMIN_EMAIL, BOTH_MODERATOR_AND_ADMIN_USERNAME)
 
-        # Navigate to any page. The role is not set.
-        self.testapp.get('/').mustcontain(no=['/moderator', '/admin'])
+        # There should be a redirect on navigating to /
+        self.assertEqual(self.testapp.get('/').status_int, 302)
```

Comment:
Oops, sorry, I take that back. I think you still want to add a new test as described, but the original line should instead be changed to

```
    self.testapp.get('/gallery').mustcontain(no=['/moderator', '/admin'])
```

The point of those lines was to check that the correct URLs show in the page depending on the user's rights.